crates/scope/src/builder.rs

```diff
@@ -2859,11 +2859,10 @@ impl ScopeDataMapBuilder {
             }
 
             if try_declare_arguments {
-                let has_defined_arguments = parameter_has_arguments || body_has_defined_arguments;
                 let declare_arguments = if has_extra_body_var {
                     !parameter_has_arguments
                 } else {
-                    has_defined_arguments
+                    !(parameter_has_arguments || body_has_defined_arguments)
                 };
```

Comment:
maybe: `let declare_arguments = !parameter_has_arguments && (has_extra_body_var || !body_has_defined_arguments);`

I do not see a similar change in https://searchfox.org/mozilla-central/source/js/src/frontend/ParseContext.cpp#612-625
Is this a bug in the original implementation?

Reply:
the condition in jsparagus side was wrong.